maven-artifact/src/test/java/org/apache/maven/artifact/versioning/ComparableVersionTest.java

```diff
@@ -346,4 +342,22 @@ public void testReuse() {
 
         assertEquals(c1, c2, "reused instance should be equivalent to new instance");
     }
+
+    /**
+     * Test <a href="https://issues.apache.org/jira/browse/MNG-7559">MNG-7559</a> edge cases
+     * 1.0.0.RC1 < 1.0.0-RC2
+     * -pfd < final, ga, release
+     * 2.0.1.MR < 2.0.1
+     * 9.4.1.jre16 > 9.4.1.jre16-preview
+     */
+    @Test
+    public void testMng7559() {
+        checkVersionsOrder("1.0.0.RC1", "1.0.0-RC2");
+        checkVersionsOrder("4.0.0.Beta3", "4.0.0-RC2");
+        checkVersionsOrder("2.3-pfd", "2.3");
+        checkVersionsOrder("2.0.1.MR", "2.0.1");
+        checkVersionsOrder("9.4.1.jre16-preview", "9.4.1.jre16");
+        checkVersionsEqual("2.0.a", "2.0.0.a"); // previously ordered, now equals
+        checkVersionsOrder("1-sp-1", "1-ga-1"); // proving website documentation right.
```

Comment:
Also there is no change here, this is weird, isn't? I mean a service pack comes *after* `GA`, no?

Reply:
i agree with you but the website documentation says otherwise. this is an edge case that may need a separate PR to fix

Reply:
According to the test `checkVersionsOrder("9.4.1.jre16-preview", "9.4.1.jre16");`
Is there a feature wish to support "filters" inside the version extension like "jre\d+"? I assume, the specification of version number must be adapted for that, because it's not the idea of "jre17" is later or better than a "jdk8" release. It's just the same release for another platform.

Reply:
i think the number are already taken into account but a unit test for will not hurt

i'll have to amend the code; meanwhile there should be a talk -maybe i should join the mailing list- about what the future yields between Maven and